exotica_core/src/kinematic_tree.cpp

```diff
@@ -279,6 +279,70 @@ void KinematicTree::BuildTree(const KDL::Tree& robot_kinematics)
 
     // Create random distributions for state sampling
     generator_ = std::mt19937(rd_());
+
+    // Add visual shapes
+    const urdf::ModelInterfaceSharedPtr& urdf = model_->getURDF();
+    std::vector<urdf::LinkSharedPtr> urdf_links;
+    urdf->getLinks(urdf_links);
+    for (urdf::LinkSharedPtr urdf_link : urdf_links)
+    {
+        if (urdf_link->visual_array.size() > 0)
+        {
+            std::shared_ptr<KinematicElement> element = tree_map_[urdf_link->name].lock();
+            int i = 0;
+            for (urdf::VisualSharedPtr urdf_visual : urdf_link->visual_array)
```

Comment:
You can reserve the number of visual elements and then below use emplace_back instead of push_back